//  Artemis metadata model
//  Protocol buffer implementation 
//  Holds all Artemis job configuration
//  Specifically, the cronus store objects
//  Menu
//  Configuration
//  Store information
//
//  Holds job state and summary information
//  All other metadata related to the actual data
//  is described in cronus

syntax = "proto3";
package artemis;

import "google/protobuf/timestamp.proto";
import "google/protobuf/duration.proto";

enum JobState {
    JOB_STARTING = 0;
    JOB_RUNNING = 1;
    JOB_FAILURE = 2;
    JOB_SUCCESS = 3;
    JOB_ABORT = 4;
    JOB_CONFIGURE = 5;
    JOB_INITIALIZE = 6;
    JOB_BOOK = 7;
    JOB_SAMPLE = 8;
    JOB_REBOOK = 9;
    JOB_EXECUTE = 10;
    JOB_FINALIZE = 11;
}

message Timer {
    string name = 1;
    float time = 2;
    float std = 3;
}

message Summary {
    uint64 processed_bytes = 1;
    repeated Timer timers = 2;
    int32 processed_ndatums = 3;
    google.protobuf.Duration job_time = 5;
}

message JobInfo {
    string name = 1;
    google.protobuf.Timestamp started = 2;
    google.protobuf.Timestamp finished = 3;
    JobState state = 4;
    string job_id = 5;
    string store_name = 6;
    string store_id = 7;
    string store_path = 8;
    string menu_id = 9;
    string config_id = 10;
    string dataset_id = 11;
    string parentset_id = 12;
    Summary summary = 13;
}
